cli/emit.rs

```diff
@@ -172,16 +172,43 @@ impl Emitter {
       ModuleSpecifier::to_file_path(specifier).unwrap(),
     )
     .await?;
-    let source_arc: Arc<str> = source_code.into();
-    let parsed_source = self
-      .parsed_source_cache
-      .remove_or_parse_module(specifier, source_arc, media_type)?;
-    let mut options = self.transpile_and_emit_options.1.clone();
-    options.source_map = SourceMapOption::None;
-    let transpiled_source = parsed_source
-      .transpile(&self.transpile_and_emit_options.0, &options)?
-      .into_source();
-    Ok(transpiled_source.text)
+    match media_type {
+      MediaType::TypeScript
+      | MediaType::Mts
+      | MediaType::Cts
+      | MediaType::Jsx
+      | MediaType::Tsx => {
+        let source_arc: Arc<str> = source_code.into();
+        let parsed_source = self
+          .parsed_source_cache
+          .remove_or_parse_module(specifier, source_arc, media_type)?;
+        // HMR doesn't work with embedded source maps for some reason, so set
+        // the option to not use them (though you should test this out because
+        // this statement is probably wrong)
+        let mut options = self.transpile_and_emit_options.1.clone();
+        options.source_map = SourceMapOption::None;
+        let transpiled_source = parsed_source
+          .transpile(&self.transpile_and_emit_options.0, &options)?
+          .into_source()
+          .into_string()?;
+        Ok(transpiled_source.text)
+      }
+      MediaType::JavaScript
+      | MediaType::Mjs
+      | MediaType::Cjs
+      | MediaType::Dts
+      | MediaType::Dmts
+      | MediaType::Dcts
+      | MediaType::Json
+      | MediaType::Wasm
+      | MediaType::TsBuildInfo
+      | MediaType::SourceMap
+      | MediaType::Unknown => {
+        // clear this specifier from the parsed source cache as it's now out of date
+        self.parsed_source_cache.free(specifier);
+        Ok(source_code)
```

Comment:
We were accidentally transpiling files that didn't need to be transpiled.

Reply:
Nice catch